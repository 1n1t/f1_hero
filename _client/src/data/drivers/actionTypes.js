/* eslint-disable import/prefer-default-export */
export const UPDATE_DRIVER_LIST = 'drivers/UPDATE_DRIVER_LIST';
