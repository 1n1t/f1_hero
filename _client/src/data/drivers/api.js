/* eslint-disable import/prefer-default-export */
import fetchApi from '../../services/api';

export const endPoints = {
	getDriverList: '/api/drivers'
};

export const getDriverList = () => fetchApi(endPoints.getDriverList);
