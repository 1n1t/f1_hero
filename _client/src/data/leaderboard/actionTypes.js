/* eslint-disable import/prefer-default-export */
export const UPDATE_SCORES = 'leaderboard/UPDATE_SCORES';
