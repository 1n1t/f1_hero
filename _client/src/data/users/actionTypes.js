/* eslint-disable import/prefer-default-export */
export const UPDATE_CURRENT_USER = 'users/UPDATE_CURRENT_USER';
