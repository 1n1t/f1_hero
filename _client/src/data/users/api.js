/* eslint-disable import/prefer-default-export */
import fetchApi from '../../services/api';

const endPoints = {
	getCurrentUser: '/auth/current-user'
};

export const getCurrentUser = () => fetchApi(endPoints.getCurrentUser);
