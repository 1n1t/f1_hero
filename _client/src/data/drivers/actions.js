/* eslint-disable import/prefer-default-export */
import * as api from './api';
import * as actionTypes from './actionTypes';

const updateDriverList = drivers => ({
	type: actionTypes.UPDATE_DRIVER_LIST,
	payload: drivers
});

export const getDriverList = () => async dispatch => {
	const response = await api.getDriverList();
	dispatch(updateDriverList(response.data.drivers));
};
