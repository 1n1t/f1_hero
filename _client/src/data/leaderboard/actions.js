/* eslint-disable import/prefer-default-export */
import * as api from './api';
import * as actionTypes from './actionTypes';

const getScores = scores => ({
	type: actionTypes.GET_SCORES,
	payload: scores
});

export const getUsersScores = () => async dispatch => {
	const response = await api.getUsersScores();
	dispatch(getScores(response.data.leaderboard));
};
