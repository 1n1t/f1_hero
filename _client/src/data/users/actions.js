/* eslint-disable import/prefer-default-export */
import * as api from './api';
import * as actionTypes from './actionTypes';

export const updateCurrentUser = user => ({
	type: actionTypes.UPDATE_CURRENT_USER,
	payload: user
});

export const getCurrentUser = () => async dispatch => {
	const response = await api.getCurrentUser();
	dispatch(updateCurrentUser(response.data.user));
};
