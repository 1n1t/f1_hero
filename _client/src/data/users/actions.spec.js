import usersSample from 'JestClient/samples/users';
import t from 'JestClient/actionTester';
import { getCurrentUser, updateCurrentUser } from './actions';
import { UPDATE_CURRENT_USER } from './actionTypes';
import { endPoints } from './api';

it(`dispatches "${UPDATE_CURRENT_USER}" action with correct payload`, () => {
	expect(updateCurrentUser(usersSample.current)).toMatchSnapshot();
});

it('fetches data from api and create correct action creator', done => {
	const dispatchMock = jest.fn();

	const assertions = (dispatch, request) => {
		expect(request.url).toEqual(endPoints.getCurrentUser);
		expect(dispatch).toHaveBeenCalledTimes(1);
		expect(dispatch).toHaveBeenCalledWith(
			updateCurrentUser(usersSample.current)
		);
		done();
	};

	t({
		action: getCurrentUser,
		apiResponse: { user: usersSample.current },
		dispatch: dispatchMock,
		assertions
	});
});
