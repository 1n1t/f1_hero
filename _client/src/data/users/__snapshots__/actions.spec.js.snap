// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dispatches "users/UPDATE_CURRENT_USER" action with correct payload 1`] = `
Object {
  "payload": Object {
    "username": "John Doe",
  },
  "type": "users/UPDATE_CURRENT_USER",
}
`;
